### PR TITLE
Unpin tox version in Beam code base

### PR DESCRIPTION
**Pull request by AnandInguva**
Leftover work of https://github.com/apache/beam/issues/24603

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @damccorm

**Review by damccorm — COMMENTED** [review]
We may end up wanting to pin to a 4.x version (probably latest), but I don't see any obvious problems with unpinning entirely - I just know historically that's what we've done.

@tvalentyn any chance you have context on why that's been done?

**Comment by damccorm** [comment]
Looks like this is causing CI to fail as well.

**Comment by AnandInguva** [comment]
> Looks like this is causing CI to fail as well.

yes, I will take a look. I will ping on the PR once they are resolved

**Comment by tvalentyn** [comment]
> @tvalentyn any chance you have context on why that's been done?

I don't remember, I think tox version shouldn't be configured in ad-hoc places like that, and I would support controlled upgrades to next major version.

Other places:
```
:beam$ rg " tox==" .
./buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
2760:                "pip install --retries 10 --upgrade tox==3.20.1 -r ${project.rootDir}/sdks/python/build-requirements.txt"

./sdks/python/apache_beam/io/hdfs_integration_test/Dockerfile
33:RUN pip install --no-cache-dir tox==3.11.1 -r sdks/python/build-requirements.txt

./sdks/python/apache_beam/io/azure/integration_test/Dockerfile
35:RUN pip install --no-cache-dir tox==3.11.1 -r sdks/python/build-requirements.txt
:beam$ 
```

**Comment by tvalentyn** [comment]
I searched around and it doesn't seem common  to specify tox upper bound requirements in codebases.

**Comment by tvalentyn** [comment]
> Looks like this is causing CI to fail as well.

> yes, I will take a look. I will ping on the PR once they are resolved

not seeing tox-related errors.

**Comment by tvalentyn** [comment]
FWIW, tox gives a headsup about breaking changes: 

```
 8839 test_1     | WARNING: test command found but not installed in testenv           
 8840 test_1     |   cmd: /bin/bash                                                   
 8841 test_1     |   env: /app/sdks/python/target/.tox/azure_integration_test         
 8842 test_1     | Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
 8843 test_1     |                                                                    
 8844 test_1     | DEPRECATION WARNING: this will be an error in tox 4 and above!     
 8845 test_1     | ___________________________________ summary ______________________________
```

**Comment by AnandInguva** [comment]
> > Looks like this is causing CI to fail as well.
> 
> > yes, I will take a look. I will ping on the PR once they are resolved
> 
> not seeing tox-related errors.

it was tox error. `/bin/bash` needs to be added to allowlist_externals list. Lets see if it resolves the error.

**Comment by AnandInguva** [comment]
Run Python_Coverage PreCommit

File: sdks/python/tox.ini [hunk 1]
```diff
@@ -40,6 +40,8 @@ allowlist_externals =
   time
   bash
   rm
+  /bin/sh
+  curl
```
Comment (AnandInguva) [inline sdks/python/tox.ini:44]:
```suggestion
  curl
  chmod
```

File: sdks/python/tox.ini [hunk 2]
```diff
@@ -40,6 +40,9 @@ allowlist_externals =
...
```
Comment (AnandInguva) [inline sdks/python/tox.ini:44]:
```suggestion
  curl
  ./codecov
```

**Comment by AnandInguva** [comment]
R: @tvalentyn

**Comment by AnandInguva** [comment]
Run Python_Integration PreCommit

**Comment by AnandInguva** [comment]
Run Python_Runners PreCommit

**Comment by liferoad** [comment]
I think pining this to a specific version is better.

**Comment by AnandInguva** [comment]
> I think pining this to a specific version is better.

I followed the pattern of github tests where we didn't pin the version https://github.com/apache/beam/blob/70eae86fc35803fd8f23e7e8c3cfda0250266d86/.github/workflows/python_tests.yml#L121.

I would like to be consistent across the code base. we can do `tox<5.0.0` everywhere in the code base

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**